craft_store/errors.py

```diff
@@ -0,0 +1,82 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Craft Store errors."""
+
+import logging
+
+import requests
+import urllib3  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class CraftStoreError(Exception):
+    """Base class error for craft-store.
+
+    :param brief: Brief description of error.
+
+    :ivar brief: Brief description of error.
+    """
+
+    def __init__(self, brief: str) -> None:
+        super().__init__()
+        self.brief = brief
+
+    def __str__(self) -> str:
+        return self.brief
```

Comment:
It looks you're using `brief` for the same thing Exception uses `message`: for a simple description of the error (and you're initiating Exception without a message).

Why not just use the normal Exception mechanism here?

I mean, here just do

```
class CraftStoreError(Exception):
    """Docstring."""
```

an below just pass the brief description without `brief=`.

Reply:
Sure, I was just following the format in https://github.com/canonical/craft-parts/blob/main/craft_parts/errors.py#L24 with the difference that I did not implement resolution or details. But indeed sounds like message could be the brief

Reply:
I have update to use the Exception's message in https://github.com/canonical/craft-store/pull/1/commits/fdb6c054eea692e19fb3bc9dd65a02a1b04fd926 ... we can look into expanding this in the future if necessary.